From 94cd2a0cd4ac4a736b13377da4fed7189c28246b Mon Sep 17 00:00:00 2001
From: Jinoh Kang <jinoh.kang.kr@gmail.com>
Date: Tue, 18 Feb 2025 02:27:20 +0900
Subject: [PATCH 295/365] ntdll: Remove redundant LDR_DONT_RESOLVE_REFS checks
 before calling process_attach().

This is now checked by process_attach() itself.
---
 dlls/ntdll/loader.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/dlls/ntdll/loader.c b/dlls/ntdll/loader.c
index 521927152cc..bf6db99ce59 100644
--- a/dlls/ntdll/loader.c
+++ b/dlls/ntdll/loader.c
@@ -953,8 +953,7 @@ static FARPROC find_forwarded_export( HMODULE module, const char *forward, LPCWS
     {
         WINE_MODREF *imp = get_modref( module );
         TRACE( "delay loading %s for '%s'\n", debugstr_w(mod_name), forward );
-        if (load_dll( load_path, mod_name, 0, &wm, imp->system ) == STATUS_SUCCESS &&
-            !(wm->ldr.Flags & LDR_DONT_RESOLVE_REFS))
+        if (load_dll( load_path, mod_name, 0, &wm, imp->system ) == STATUS_SUCCESS)
         {
             if (!imports_fixup_done && importer)
             {
@@ -3628,7 +3627,7 @@ NTSTATUS WINAPI DECLSPEC_HOTPATCH LdrLoadDll(LPCWSTR path_name, DWORD flags,
 
     nts = load_dll( path_name, dllname ? dllname : libname->Buffer, flags, &wm, FALSE );
 
-    if (nts == STATUS_SUCCESS && !(wm->ldr.Flags & LDR_DONT_RESOLVE_REFS))
+    if (nts == STATUS_SUCCESS)
     {
         nts = process_attach( wm->ldr.DdagNode, NULL );
         if (nts != STATUS_SUCCESS)
-- 
2.50.1

